File: Easy/2460. Apply Operations to an Array/JavaScript/main.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var applyOperations = function (nums) {
  let n = nums.length;

  // Step 1: Apply the given operation sequentially
  for (let i = 0; i < n - 1; i++) {
    if (nums[i] === nums[i + 1]) {
      nums[i] *= 2;
      nums[i + 1] = 0;
    }
  }

  // Step 2: Shift all zeros to the end
  let result = []
  let zeroCount = 0;

  for (let num of nums) {
    if (num !== 0) {
      result.push(num)
    } else {
      zeroCount++
    }
  }

  // Append the zeros at the end
  while (zeroCount--) {
    result.push(0)
  }

  return result;
};

File: Easy/2460. Apply Operations to an Array/Rust/main.rs
impl Solution {
    pub fn apply_operations(nums: Vec<i32>) -> Vec<i32> {
        let mut nums = nums;
        let n = nums.len();

        for i in 0..n - 1 {
            if nums[i] == nums[i + 1] {
                nums[i] *= 2;
                nums[i + 1] = 0;
            }
        }

        let mut result: Vec<i32> = Vec::new();
        let mut zeros = 0;

        for num in nums {
            if num == 0 {
                zeros += 1;
            } else {
                result.push(num);
            }
        }

        for _ in 0..zeros {
            result.push(0);
        }

        result
    }
}
